dlio/README.md

```diff
@@ -0,0 +1,118 @@
+Evaluation throughput standard deviation: 0.090 samples/s
+Evaluation average IO: 116.883 MB/s
+Evaluation IO standard deviation: 5.735 MB/s
+===========================================================
+```
+Let's take a closer look at it. First, information about the number of MPI ranks and processes per MPI rank used in the 
+simulation is output. Then, the same values are used to describe the training and evaluation performance, so for the 
+sake of reducing redundancy, let us consider only the first half of the results concerning the training process. Total 
+training set size is calculated as the size of all HDF5 files used for training. Accordingly, the training set size per 
+rank gives an idea of how much of the load is taken over by one MPI rank. Total training emulated compute time contains 
+information about the total time spent on compute emulation for all epochs in total, as well as training metadata time 
+and training raw read time, about which, however, it should be noted that they are not interleaved and measure the time 
+of execution of `H5Fopen`, `H5Dget_space`, `H5Screate_simple`, `H5Sclose` and `H5Dread` commands respectively. Training 
+average raw read rate is calculated as training set size per rank divided by training raw read time. Observed training 
+completion time includes all the time spent on the training process, among other things including resource allocation 
+and computation simulation. Observed average training rate is equal to training set size per rank divided by the 
+difference of observed training completion time and total training emulated compute time, thus showing the data reading 
+rate without taking into account emulation costs. Training average throughput and training throughput standard deviation 
+give an indication of the number of samples from the training dataset processed in one second. Training average IO and 
+Training IO standard deviation translate these values into bytes/second by multiplying by the size of one sample.
+
+## Future work
+
+There are plans to add more configuration options for the extension in the future to increase its flexibility:
+- Add settings for Subfiling VFD. Currently, the default settings are used.
+- Add more features from [DLIO Benchmark](https://github.com/argonne-lcf/dlio_benchmark) such as resizable records.
+- Analyze and add support for other ml frameworks and data loaders. For example, DALI.
+- Add support for prefetching.
+- Expand the ability to randomly shuffle samples. At the moment, it is not possible to shuffle only samples in each file
+without changing the order of the files for training.
+- Add more compression filters and thus support different compression algorithms for HDF5 data.
+- Add support for drop_last customization. Currently, by default, all batches left after MPI ranks distribution are not processed.
+- Replace the use of `fork()` with `MPI_Comm_spawn()` when creating new processes, as using `fork()` with MPI may be unsafe
```

Comment:
as this is unsafe, why are we using it today?

Reply:
Because it is `fork()` that is [used in PyTorch for multiprocessing](https://pytorch.org/docs/stable/data.html#platform-specific-behaviors). At this point, the code is too simple to cause any problems - as we can see, it compiles and runs on both CLAIX and Perlmutter. The question of whether using `fork()` in our case is safe needs to be investigated further.